From ed65910c794538ff24e202be0925833ffbd96db3 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Fri, 31 Aug 2018 11:27:29 +0800
Subject: [PATCH 4618/5242] MLK-19318 drm/imx: lcdif: avoid log print when no
 device attached

commit  0fbeb07f8e91c6658c97c601688f6b168c74cc15 from
https://source.codeaurora.org/external/imx/linux-imx.git

In the 'lcdif_crtc_atomic_check()', when the 'bus_format' is
zero which means that there is no valid display peripherals
attached to LCDIF, return directly to avoid below error log
to make noises, since the error log is not cared in this case.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit 25d2b80f637af06094f56c60d46404af3b7ff381)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/lcdif/lcdif-crtc.c |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/gpu/drm/imx/lcdif/lcdif-crtc.c b/drivers/gpu/drm/imx/lcdif/lcdif-crtc.c
index c9a617e..457483e 100644
--- a/drivers/gpu/drm/imx/lcdif/lcdif-crtc.c
+++ b/drivers/gpu/drm/imx/lcdif/lcdif-crtc.c
@@ -94,6 +94,13 @@ static int lcdif_crtc_atomic_check(struct drm_crtc *crtc,
 	struct lcdif_crtc *lcdif_crtc = to_lcdif_crtc(crtc);
 	struct imx_crtc_state *imx_crtc_state = to_imx_crtc_state(state);
 
+	/* return directly when no devices attached
+	 * to LCDIF to avoid below error messsage to
+	 * make noises in this case.
+	 */
+	if (!imx_crtc_state->bus_format)
+		return -EINVAL;
+
 	/* check the requested bus format can be
 	 * supported by LCDIF CTRC or not
 	 */
-- 
1.7.9.5

